File: tests/fixtures/conformance/typescript/classes.ts
// Feature: a class with a constructor and methods (owns edges, method kinds).

export class Counter {
  private value: number;

  constructor(start: number) {
    this.value = start;
  }

  increment(): number {
    this.value += 1;
    return this.value;
  }

  reset(): void {
    this.value = 0;
  }
}

File: tests/fixtures/conformance/csharp/classes.cs
// Feature: a class with a constructor, an auto-property, and methods that call each
// other — the bread-and-butter C# type. Exercises symbols (class/constructor/property/
// method), owns edges, and an intra-class call edge.
using System;

namespace Shop.Billing
{
    public class Invoice
    {
        public Invoice(string customer)
        {
            Customer = customer;
        }

        public string Customer { get; set; }

        public decimal Total(decimal rate)
        {
            return Apply(rate);
        }

        private decimal Apply(decimal rate)
        {
            return rate;
        }
    }
}
